app/lib/features/news/widgets/news_item_slide/text_slide.dart

```diff
@@ -4,42 +4,91 @@ import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_riverpod/flutter_riverpod.dart';
 
-class TextSlide extends ConsumerWidget {
+class TextSlide extends ConsumerStatefulWidget {
   final NewsSlide slide;
   final Color bgColor;
   final Color fgColor;
+  final PageController pageController;
 
   const TextSlide({
     super.key,
     required this.slide,
     required this.bgColor,
     required this.fgColor,
+    required this.pageController,
   });
 
   @override
-  Widget build(BuildContext context, WidgetRef ref) {
-    final slideContent = slide.msgContent();
+  ConsumerState<TextSlide> createState() => _TextSlideState();
+}
+
+class _TextSlideState extends ConsumerState<TextSlide> {
+  late ScrollController _scrollController;
+
+  @override
+  void initState() {
+    super.initState();
+    _scrollController = ScrollController();
+    _scrollController.addListener(_onScroll);
+  }
+
+  void _onScroll() {
+    final topPosition = _scrollController.position.pixels <= 0;
+    final outOfRange = _scrollController.position.outOfRange;
+    final offset = _scrollController.offset;
+    final maxScrollExtent = _scrollController.position.maxScrollExtent;
+    if (topPosition && outOfRange && offset < -150) {
+      widget.pageController.previousPage(
+        duration: const Duration(milliseconds: 300),
+        curve: Curves.ease,
+      );
+    } else if (!topPosition && outOfRange && offset > maxScrollExtent + 150) {
+      widget.pageController.nextPage(
+        duration: const Duration(milliseconds: 300),
+        curve: Curves.ease,
+      );
+    }
+  }
```

Comment:
yeah not able to review this for sure ... so ... I just trust this is set up as we want ...

Reply:
Yes it is all about scroll behaviour things.